src/utils/formatNumber.ts

```diff
@@ -193,3 +193,22 @@ export const formatCurrency = (number: string | number, currency: string): strin
 
   return format(number, currencyFormatter, minimum)
 }
+
+export const localeNumberFormatter = (amount: number | string): string => {
+  // get locale decimal separator
+  const number = 123456.789
```

Comment:
Using the arbitrary value confused me at first. I would suggest using [`Intl.NumberFormat.prototype.formatToParts`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/formatToParts) instead as you can directly get the separators from `amount` directly.